src/xo-app/backup/restore/index.js

```diff
@@ -120,44 +119,49 @@ class _ModalBody extends Component {
   }
 
   _getSrPredicate = createSelector(
-    () => this.state.sr,
+    () => this.state.mainSr,
     () => this.state.mapVdisSrs,
-    (defaultSr, mapVdisSrs) => sr =>
-      sr !== defaultSr &&
+    (mainSr, mapVdisSrs) => sr =>
       isSrWritable(sr) &&
-      defaultSr.$pool === sr.$pool &&
-      areSrsCompatible(defaultSr, sr) &&
+      mainSr.$pool === sr.$pool &&
+      areSrsCompatible(mainSr, sr) &&
       every(mapVdisSrs, selectedSr => selectedSr == null || areSrsCompatible(selectedSr, sr))
   )
 
-  _onChangeDefaultSr = event => {
-    const oldSr = this.state.sr
-    const newSr = getEventValue(event)
+  _onChange = props => {
+    if (props.mainSr !== undefined) {
+      const newMainSr = props.mainSr
+      const oldMainSr = this.state.mainSr
 
-    if (oldSr == null || newSr == null || oldSr.$pool !== newSr.$pool) {
-      this.setState({
-        mapVdisSrs: {}
-      })
-    } else if (!newSr.shared) {
-      const mapVdisSrs = {...this.state.mapVdisSrs}
-      forEach(mapVdisSrs, (sr, vdi) => {
-        if (sr != null && newSr !== sr && sr.$container !== newSr.$container && !sr.shared) {
-          delete mapVdisSrs[vdi]
-        }
-      })
-      this.setState({
-        mapVdisSrs
-      })
+      if (oldMainSr == null || newMainSr == null || oldMainSr.$pool !== newMainSr.$pool) {
+        this.setState({
+          mapVdisSrs: {}
+        })
+      } else if (!newMainSr.shared) {
+        const mapVdisSrs = {...this.state.mapVdisSrs}
+        forEach(mapVdisSrs, (sr, vdi) => {
+          if (sr != null && newMainSr !== sr && sr.$container !== newMainSr.$container && !sr.shared) {
+            delete mapVdisSrs[vdi]
+          }
+        })
+        this.setState({mapVdisSrs})
+      }
     }
 
-    this.setState({
-      sr: newSr
-    })
+    this.setState(props)
```

Comment:
Can `props` have both `mainSr` and `mapVdisSrs` properties at the same time?
- if yes: this `setState` would override the previous ones' `mapVdisSrs` value
- if not: wouldn't it be a good thing to assume it could be possible to prevent issues with future code changes?